tests/Support/SupportLazyCollectionIsLazyTest.php

```diff
@@ -977,6 +977,17 @@ public function testSomeIsLazy()
         });
     }
 
+    public function testSoleIsLazy()
+    {
+        $data = $this->make([['a' => 1], ['a' => 2], ['a' => 3]]);
+
+        $this->assertEnumeratesCollection($data, 3, function ($collection) {
+            $collection->sole(function ($item) {
+                return $item['a'] === 2;
+            });
+        });
+    }
```

Comment:
```suggestion
    public function testSoleIsLazy()
    {
        $this->assertEnumeratesOnce(function ($collection) {
            $collection->sole(function ($item) {
                return $item === 1;
            });
        });

        $this->assertEnumerates(4, function ($collection) {
            try {
                $collection->sole(function ($item) {
                    return $item % 2 === 0;
                });
            } catch (MultipleItemsFoundException $e) {
                //
            }
        });
    }
```

Be sure to also import `MultipleItemsFoundException`.

Reply:
Thanks for the suggestion again, I've just pushed up the new test changes :)